src/debug_menu.cpp

```diff
@@ -3733,30 +3733,43 @@ static void spawn_npc()
 
 static void spawn_named_npc()
 {
-    const std::string input = string_input_popup()
-                              .title( _( "Enter NPC template" ) )
-                              .width( 20 )
-                              .query_string();
+    uilist npc_menu;
 
-    if( input.empty() ) {
-        return;
+    npc_menu.text = _( "Choose NPC:" );
+
+    std::vector<npc_template_id> npc_list;
+
+    for( const auto &e : npc_template::get_npc_templates() ) {
+        npc_list.emplace_back( e.first );
     }
 
-    const npc_template_id npc_template = npc_template_id( input );
-    if( !npc_template.is_valid() ) {
-        popup( "Invalid template id" );
-        return;
+    std::sort( npc_list.begin(), npc_list.end(), localized_compare );
```

Comment:
this sorting doesn't seem to work, i suppose because the string_id<npc_template> doesn't do operator< correctly for strings?